### PR TITLE
[14.0] website: fix res.config settings

### PR DESCRIPTION
**Pull request by yelizariev**
[FIX] website: allow edit robots.txt to non-first website
Previous implementation always updated *current website*, which is not
always the same as the value from `res.config.settings`.

[FIX] website: allow submit sitemap to non-first website

### CODE REVIEW

File: addons/website/wizard/website_robots.py [hunk 1]
```diff
@@ -7,8 +7,13 @@ class WebsiteRobots(models.TransientModel):
     _name = "website.robots"
     _description = "Robots.txt Editor"
 
-    content = fields.Text(default=lambda s: s.env['website'].get_current_website().robots_txt)
+    content = fields.Text()
+    website_id = fields.Many2one('website')
```
Comment (yelizariev) [inline addons/website/wizard/website_robots.py:11]: The new field makes `ci/template` check red and requires special intervention from runbot/upgrade team as I remember

Comment (rdeodoo) [reply addons/website/wizard/website_robots.py:11]: Remove this new field

File: addons/website/wizard/website_robots.py [hunk 2]
```diff
@@ -7,8 +7,13 @@ class WebsiteRobots(models.TransientModel):
     _name = "website.robots"
     _description = "Robots.txt Editor"
 
-    content = fields.Text(default=lambda s: s.env['website'].get_current_website().robots_txt)
+    content = fields.Text()
```
Comment (rdeodoo) [inline addons/website/wizard/website_robots.py:10]: leave this untouched, just alter `website_id` in context so `get_current_website` will return the correct one (it depend of context)

File: addons/website/models/res_config_settings.py [hunk 3]
```diff
@@ -175,6 +175,7 @@ def action_open_robots(self):
             'type': 'ir.actions.act_window',
             "views": [[False, "form"]],
             'target': 'new',
+            'context': {'default_website_id': self.website_id.id}
```
Comment (rdeodoo) [inline addons/website/models/res_config_settings.py:178]:
```suggestion
            'context': {'website_id': self.website_id.id}
```
remove everything else from the PR (except the js part)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Just add `website_id` in context and `get_current_website` will act accordingly

File: addons/website/wizard/website_robots.py (hunk 1 above)
Comment (yelizariev) [reply addons/website/wizard/website_robots.py:11]: I've tried, but it doesn't work if session has `force_website_id`

https://github.com/odoo/odoo/blob/d3feb26c8923cfde60894058f06dc2c2a8be05f5/addons/website/models/website.py#L568-L574

Comment (rdeodoo) [reply addons/website/wizard/website_robots.py:11]: Then pop the force website id from session in `action_open_robots` :thinking:
Or force it to selected website.. as we do when you click on go to website or install theme in settings..
`self.website_id._force()`

Comment (rdeodoo) [reply addons/website/wizard/website_robots.py:11]: And you won't need to alter the context as a bonus ;-)

**Review by rdeodoo — APPROVED** [review]

File: addons/website/wizard/website_robots.py (hunk 1 above)
Comment (rdeodoo) [reply addons/website/wizard/website_robots.py]: You are welcome, indeed it is ! LGTM now

**Comment by JKE-be** [comment]
ty

@robodoo r+

**Comment by JKE-be** [comment]
@robodoo rebase-ff

**Pull request closed without merging**